### PR TITLE
New ruleset endpoint

### PR DESCRIPTION
**Pull request by tealeg**
This PR implements a handler for POSTs to the ruleset endpoint from the UI, in accordance with the issue #31 

On receipt of POST to the `/ui/i/rulesets` endpoint we attempt to:
- unmarshal the payload into a struct, 
- convert that structure into a `regula.Ruleset` and its `regula.rule.Rule`(s).  This step involves the symbolic expression parser parsing each rule.  
- Return either a StatusCreated (HTTP 201) or a StatusBadRequest (HTTP 400)

Mostly of this branch is self evident, but I did restructure the request handler to now cope with POST as well as GET requests.

### CODE REVIEW

File: ui/handler.go [hunk 1]
```diff
@@ -75,8 +88,36 @@ func newInternalHandler(service store.RulesetService) http.Handler {
 	return &mux
 }
 
-// Returns an http handler that lists all existing rulesets paths.
-func (h *internalHandler) rulesetsHandler() http.Handler {
+// handleNewRulesetRequest consumes a POST to the ruleset endpoint and
+// attempts to create a new Ruleset from that data.
+func (h *internalHandler) handleNewRulesetRequest(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
```
Comment (asdine) [inline ui/handler.go]: :tulip: Small tip: You can also use `json.NewDecoder(r.Body).Decode`

File: ui/handler.go [hunk 2]
```diff
@@ -75,8 +88,36 @@ func newInternalHandler(service store.RulesetService) http.Handler {
...
+	if err != nil {
+		writeError(w, r, err, http.StatusBadRequest)
```
Comment (asdine) [inline ui/handler.go]: You forgot a return after this line

File: ui/handler.go [hunk 3]
```diff
@@ -85,27 +126,203 @@ func (h *internalHandler) rulesetsHandler() http.Handler {
...
+	Path      string    `json:"path"`
+	Signature signature `json:"signature"`
+	Rules     []rule    `json:"rules"`
+}
 
-		// run the loop at least once, no matter of the value of token
-		for i := 0; i == 0 || opt.ContinueToken != ""; i++ {
-			list, err := h.service.List(r.Context(), "", &opt)
+// convertParams takes a slice of param, unmarshalled from a
+// newRulesetRequest, and returns an equivalent sexpr.Paramaters map.
```
Comment (asdine) [inline ui/handler.go]:
```suggestion
// newRulesetRequest, and returns an equivalent sexpr.Parameters map.
```

**Comment by asdine** [comment]
:warning: The build is failing

**Comment by tealeg** [comment]
:up: @asdine

**Pull request merged**